src/constants.rs

```diff
@@ -13,24 +13,57 @@ pub struct Paths {
     cas_store: ContentAddressable,
 }
 
+/// Everything that can happen when creating `Paths`.
+/// Mostly filesystem access problems.
+#[derive(Debug)]
+pub enum PathsInitError {
+    /// The `gc_root_dir` creation failed.
+    #[allow(missing_docs)]
+    GcRootsDirectoryCantBeCreated {
```

Comment:
I suggest
- `InitPathsError::CreateGcRootsDir`
- `InitPathsError::CreateSocketDir`
- `InitPathsError::CreateCas`

I think this fits better with how errors are named in the [standard library](https://doc.rust-lang.org/std/all.html?search=error).

Reply:
I’d like for them to be as descriptive as possible, so the user knows what the problem was. `CreateGcRootsDir` doesn’t really give the same air as `GcRootsDirectoryCantBeCreated`. (They are not really intended to be consumed programmatically anyway.)